data-protocols/ids/ids-api-multipart-endpoint-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/handler/ArtifactRequestHandler.java

```diff
@@ -78,102 +78,96 @@ public boolean canHandle(@NotNull MultipartRequest multipartRequest) {
 
+        // Read request payload, which contains the data destination and an optional secret
         ArtifactRequestMessagePayload artifactRequestMessagePayload;
         try {
-            artifactRequestMessagePayload =
-                    objectMapper.readValue(multipartRequest.getPayload(), ArtifactRequestMessagePayload.class);
+            artifactRequestMessagePayload = objectMapper.readValue(multipartRequest.getPayload(), ArtifactRequestMessagePayload.class);
         } catch (IOException e) {
-            return createBadParametersErrorMultipartResponse(artifactRequestMessage);
+            return createMultipartResponse(badParameters(multipartRequest.getHeader(), connectorId));
         }
 
-        var dataAddress = artifactRequestMessagePayload.getDataDestination();
+        var dataDestination = artifactRequestMessagePayload.getDataDestination();
 
+        // Read request message properties
         Map<String, String> props = new HashMap<>();
-        if (artifactRequestMessage.getProperties() != null) {
-            artifactRequestMessage.getProperties().forEach((k, v) -> props.put(k, v.toString()));
+        if (message.getProperties() != null) {
+            message.getProperties().forEach((k, v) -> props.put(k, v.toString()));
         }
 
+        // Get webhook address of requesting connector from message properties
         var idsWebhookAddress = Optional.ofNullable(props.remove(IDS_WEBHOOK_ADDRESS_PROPERTY))
                 .map(Object::toString)
                 .orElse(null);
         if (StringUtils.isNullOrBlank(idsWebhookAddress)) {
             var msg = "Ids webhook address is invalid";
             monitor.debug(String.format("%s: %s", getClass().getSimpleName(), msg));
-            return createBadParametersErrorMultipartResponse(artifactRequestMessage, msg);
+            return createMultipartResponse(badParameters(multipartRequest.getHeader(), connectorId));
         }
 
         // NB: DO NOT use the asset id provided by the client as that can open aan attack vector where a client references an artifact that
         //     is different from the one specified by the contract
 
         var dataRequest = DataRequest.Builder.newInstance()
-                .id(artifactRequestMessage.getId().toString())
+                .id(message.getId().toString())
                 .protocol(Protocols.IDS_MULTIPART)
-                .dataDestination(dataAddress)
+                .dataDestination(dataDestination)
                 .connectorId(connectorId)
                 .assetId(contractAgreement.getAssetId())
                 .contractId(contractAgreement.getId())
                 .properties(props)
                 .connectorAddress(idsWebhookAddress)
                 .build();
 
-        var result = transferProcessManager.initiateProviderRequest(dataRequest);
+        // Initiate a transfer process for the request
+        var transferInitiateResult = transferProcessManager.initiateProviderRequest(dataRequest);
```

Comment:
note for the future (after project structure review), here the `TransferProcessService` should be used instead of the manager

Reply:
Hey, @ronjaquensel, this aligns with our discussion to encapsulate some of the policy evaluation as well.

Reply:
@ronjaquensel Can we either create an issue or add a TODO for this comment and the one regarding the `ContractNegotiationService`?

Reply:
good point, I added TODOs.